### PR TITLE
Listing screenings

### PR DESCRIPTION
**Pull request by danyflorczak**
- Created models and schema for Halls, Movies, Screenings
- Created CRUD views for Halls, Movies, Screenings
- Created many-to many association between models 
- For now all views ( new , index , show etc. ) are left for halls and movies as soon as homepage will be created those views that are not necessary will be deleted
- Created basic validation

### CODE REVIEW

File: db/migrate/20220913112745_create_movies.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateMovies < ActiveRecord::Migration[7.0]
+  def change
+    create_table :movies do |t|
+      t.string :title, null: false
+      t.text :description
+      t.float :duration
```
Comment (lilka) [inline db/migrate/20220913112745_create_movies.rb:6]: Consider using the `interval` type here. https://www.bigbinary.com/blog/rails-6-1-adds-postgresql-interval-data-type

File: db/migrate/20220913112102_create_screenings.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateScreenings < ActiveRecord::Migration[7.0]
+  def change
+    create_table :screenings do |t|
+      t.datetime :start_time, null: false
+      t.datetime :end_time,null: false
```
Comment (lilka) [inline db/migrate/20220913112102_create_screenings.rb:5]: add space after comma ;)

File: app/controllers/movies_controller.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    end
+  
+    private
+      def set_movie
+        @movie = Movie.find(params[:id])
+      end
+  
+      def movie_params
+        params.require(:movie).permit(:title,:description, :duration)
+      end
```
Comment (lilka) [inline app/controllers/movies_controller.rb:47]: add space after the comma

File: app/views/screenings/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+  
+  <div class="field">
+    <%= form.label "Movie:" %>
+    <%= form.select(:movie_id, Movie.all.collect {|c| [c.title, c.id ]}, { :include_blank => 'Wybierz' }) %>
+  </div>
+
```
Comment (lilka) [inline app/views/screenings/_form.html.erb:18]: I would  suggest sticking to one language in the app

**Review by lilka — COMMENTED** [review]
Generally looks good to me 👍 . Try to create smaller PRs.

File: test/fixtures/movies.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
+# This model initially had no columns defined. If you add columns to the
+# model remove the "{}" from the fixture names and add the columns immediately
+# below each fixture, per the syntax in the comments below
+#
+one: {}
```
Comment (h4ppyr0gu3) [inline test/fixtures/movies.yml]: could delete this file i think

File: test/models/movie_test.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class MovieTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (h4ppyr0gu3) [inline test/models/movie_test.rb]: can delete this file as will use specs

File: test/models/screening_test.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class ScreeningTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (h4ppyr0gu3) [inline test/models/screening_test.rb]: as above

File: app/controllers/halls_controller.rb [hunk 8]
```diff
@@ -0,0 +1,56 @@
+class HallsController < ApplicationController
+    before_action :set_hall, only: %i[ show edit update destroy ]
+  
+    # GET /halls
```
Comment (h4ppyr0gu3) [inline app/controllers/halls_controller.rb]: these comments are unnecessary

**Review by h4ppyr0gu3 — APPROVED** [review]
looks good but as Tobiasz said, try to keep pull request smaller

**Pull request merged**